src/halmos/__main__.py

```diff
@@ -1413,6 +1424,47 @@ def run_tests(
     test_results = []
     debug_config = args.debug_config
 
+    # pretty print the target contracts and functions for invariant tests
+    has_invariant_tests = any(funsig.startswith("invariant_") for funsig in funsigs)
+    if has_invariant_tests:
+        try:
+            target_contracts = resolve_target_contracts(ctx, setup_ex)
+        except HalmosException:
+            # no target contracts found, let the downstream tests fail
+            target_contracts = []
+
+        panel_content = []
+        for target_contract in target_contracts:
+            code = setup_ex.code[target_contract]
+            contract_name = code.contract_name
+            filename = code.filename
+            contract_json = BuildOut().get_by_name(contract_name, filename)
+
+            addr_str = rendered_address(
+                target_contract, replace_with_contract_name=False
+            )
+            tree = Tree(f"[bold]{filename}:{contract_name}[/bold] @ {addr_str}")
+
+            target_selectors = resolve_target_selectors(
+                ctx, target_contract, contract_json
+            )
+            for fun_sig, _ in target_selectors:
+                tree.add(f"{fun_sig}")
+
+            panel_content.append(tree)
+            panel_content.append(Text(""))
+
+        if panel_content:
+            # pop the last newline
+            panel_content.pop()
+            panel = Panel(
+                Group(*panel_content),
+                title="Initial Invariant Target Functions",
+                border_style="blue",
+                expand=False,
+            )
+            ui.print("\n", panel, "\n")
+
```

Comment:
can we factor out this display logic into a separate function for better readability?

Reply:
yep, on it

Reply:
in [c87cefc](https://github.com/a16z/halmos/pull/516/commits/c87cefcf693bf74fef2833690f1f0bf41e2adc95)